# Apply this in the system directory where guile was installed
# (e.g., /usr/share/include)
--- libguile/error.h	1999/07/06 17:56:42	1.1
+++ libugile/error.h	1999/07/06 17:57:11	1.2
@@ -59,24 +59,24 @@
 #define SCM_NORETURN
 #endif
 
-extern void scm_error SCM_P ((SCM key, char *subr, char *message,
+extern void scm_error SCM_P ((SCM key, const char *subr, const char *message,
 			      SCM args, SCM rest)) SCM_NORETURN;
 extern SCM scm_error_scm SCM_P ((SCM key, SCM subr, SCM message,
 			      SCM args, SCM rest)) SCM_NORETURN;
 extern SCM scm_strerror (SCM err);
-extern void scm_syserror SCM_P ((char *subr)) SCM_NORETURN;
-extern void scm_syserror_msg SCM_P ((char *subr, char *message, SCM args,
+extern void scm_syserror SCM_P ((const char *subr)) SCM_NORETURN;
+extern void scm_syserror_msg SCM_P ((const char *subr, char *message, SCM args,
 				     int eno)) SCM_NORETURN;
-extern void scm_sysmissing SCM_P ((char *subr)) SCM_NORETURN;
-extern void scm_num_overflow SCM_P ((char *subr)) SCM_NORETURN;
-extern void scm_out_of_range SCM_P ((char *subr, SCM bad_value)) SCM_NORETURN;
+extern void scm_sysmissing SCM_P ((const char *subr)) SCM_NORETURN;
+extern void scm_num_overflow SCM_P ((const char *subr)) SCM_NORETURN;
+extern void scm_out_of_range SCM_P ((const char *subr, SCM bad_value)) SCM_NORETURN;
 extern void scm_wrong_num_args SCM_P ((SCM proc)) SCM_NORETURN;
-extern void scm_wrong_type_arg SCM_P ((char *subr, int pos, SCM bad_value))
+extern void scm_wrong_type_arg SCM_P ((const char *subr, int pos, SCM bad_value))
                                      SCM_NORETURN;
 extern void scm_memory_error SCM_P ((char *subr)) SCM_NORETURN;
-extern void scm_misc_error SCM_P ((char *subr, char *message, SCM args))
+extern void scm_misc_error SCM_P ((const char *subr, const char *message, SCM args))
 				 SCM_NORETURN;
-extern SCM scm_wta SCM_P ((SCM arg, char *pos, char *s_subr));
+extern SCM scm_wta SCM_P ((SCM arg, const char *pos, const char *s_subr));
 extern void scm_init_error SCM_P ((void));
 
 #endif  /* ERRORH */
--- libguile/genio.h	1999/07/06 17:57:54	1.1
+++ libguile/genio.h	1999/07/06 17:58:11	1.2
@@ -49,7 +49,7 @@
 
 
 extern void scm_putc SCM_P ((int c, SCM port));
-extern void scm_puts SCM_P ((char *str_data, SCM port));
+extern void scm_puts SCM_P ((const char *str_data, SCM port));
 extern void scm_lfwrite SCM_P ((char *ptr, scm_sizet size, SCM port));
 extern void scm_fflush SCM_P ((SCM port));
 extern int scm_getc SCM_P ((SCM port));
--- libguile/snarf.h	1999/07/06 19:20:42	1.1
+++ libguile/snarf.h	1999/07/06 19:33:07	1.2
@@ -55,9 +55,9 @@
 	static char RANAME[]=STR
 #else
 #define SCM_PROC(RANAME, STR, REQ, OPT, VAR, CFN)  \
-%%%	scm_make_gsubr (RANAME, REQ, OPT, VAR, CFN)
+%%%	scm_make_gsubr (RANAME, REQ, OPT, VAR, (SCM (*)(...))CFN)
 #define SCM_PROC1(RANAME, STR, TYPE, CFN)  \
-%%%	scm_make_subr(RANAME, TYPE, CFN)
+%%%	scm_make_subr(RANAME, TYPE, (SCM (*)(...))CFN)
 #endif
 
 #ifndef SCM_MAGIC_SNARFER
--- guile/gh.h	1999/07/06 18:51:48	1.1
+++ guile/gh.h	1999/07/06 18:52:46	1.2
@@ -103,10 +103,10 @@
 SCM gh_long2scm(long x);
 SCM gh_double2scm(double x);
 SCM gh_char2scm(char c);
-SCM gh_str2scm(char *s, int len);
-SCM gh_str02scm(char *s);
-void gh_set_substr(char *src, SCM dst, int start, int len);
-SCM gh_symbol2scm(char *symbol_str);
+SCM gh_str2scm(const char *s, int len);
+SCM gh_str02scm(const char *s);
+void gh_set_substr(const char *src, SCM dst, int start, int len);
+SCM gh_symbol2scm(const char *symbol_str);
 SCM gh_ints2scm(int *d, int n);
 SCM gh_longs2ivect(long *d, int n);
 SCM gh_ulongs2uvect(unsigned long *d, int n);
